srf_generation/source_parameter_generation/nshm2022_to_realisation.py

```diff
@@ -0,0 +1,434 @@
+
+    Returns
+    -------
+    (np.ndarray, np.ndarray)
+        The closest pair of points (u, v) with u in from_fault and v in
+        to_fault. The points are specified in WGS84+depth coordinates.
+    """
+
+    from_fault_point_nztm, to_fault_point_nztm = closest_points_between_faults(
+        from_fault, to_fault
+    )
+    from_fault_point_wgsdepth = qcore.coordinates.nztm_to_wgs_depth(
+        from_fault_point_nztm.reshape((1, -1))
+    ).ravel()
+    to_fault_point_wgsdepth = qcore.coordinates.nztm_to_wgs_depth(
+        to_fault_point_nztm.reshape((1, -1))
+    ).ravel()
+    return from_fault_point_wgsdepth, to_fault_point_wgsdepth
+
+
+def link_hypocentres(
+    rupture_causality_tree: RuptureCausalityTree, faults: list[RealisationFault]
+):
+    """Set the jumping points across faults.
+
+    Given a rupture causality tree, and the list of faults on that tree, set the
+    parent, shyp, dhyp and jump point coordinates based on the computed fault
+    jumping points. Modifies the faults array.
+
+    Parameters
+    ----------
+    rupture_causality_tree : RuptureCausalityTree
+        The rupture causality tree that specifies how the faults jump.
+    faults : list[RealisationFault]
+        The list of faults to link.
+    """
+    fault_name_map = {fault.name: fault for fault in faults}
+    for to_fault in faults:
+        fault_name = to_fault.name
+        if rupture_causality_tree[fault_name] is None:
+            shyp, dhyp = to_fault.expected_fault_hypocentre_coordinates()
+            to_fault.shyp = shyp
+            to_fault.dhyp = dhyp
+        else:
+            from_fault = fault_name_map[rupture_causality_tree[fault_name]]
+            from_fault_point, to_fault_point = (
+                compute_jump_point_hypocentre_fault_coordinates(from_fault, to_fault)
+            )
+            try:
+                to_shyp, to_dhyp = to_fault.global_coordinates_to_fault_coordinates(
+                    to_fault_point
+                )
+            except ValueError:
+                breakpoint()
+            to_fault.parent = from_fault
+            to_fault.shyp = float(to_shyp)
+            to_fault.dhyp = float(to_dhyp)
+            to_fault.parent_jump_coords = tuple(float(x) for x in from_fault_point)
+
+
+def normalise_name(name: str) -> str:
+    """Normalise a fault name for yaml file output.
+
+    Parameters
+    ----------
+    name : str
+        The name to normalise.
+
+    Returns
+    -------
+    str
+        The normalised version of the name.
+    """
+    fault_no_illegal_characters = re.sub(r" |,|:", "_", name)
+    return fault_no_illegal_characters.lower()
+
+
+def magnitude_for_fault(target_fault: RealisationFault, total_area: float) -> float:
+    """Calculate the target fault's contribution to the total magnitude of a given rupture.
+
+    Given a target fault, and the total area of a rupture the fault participates
+    in, calculation the proportion of the magnitude that the target fault would
```

Comment:
small typo "...fault participates in, _calculation_ the proportion " possibly should be _calculate_